### PR TITLE
Fix opcode mismatch

### PR DESCRIPTION
**Pull request by RoiEXLab**
Fix for #6104 and potentially other issues.
There is one thing that could potentially (in theory) still be an issue, but at this point it's highly unlikely it actually exists:
Interfaces that inherit their methods from `IRemote` and `IDelegate` don't specify an op-code for the methods defined in `IDelegate` and subinterfaces, just the ones specified in subinterfaces of `IRemote`. This decision was intentional and up until this point it does ot seem like delegate-only methods are ever invoked over the network, the client side would report this issue already.

## Functional Changes

[] New map or map update
[] New Feature
[] Feature update or enhancement
[] Feature Removal
[] Code Cleanup or refactor
[] Configuration Change
[x] Problem fix:  #6104
[] Other:   

## Testing

I verified #6104 could no longer be reproduced.

### CODE REVIEW

File: game-core/src/main/java/games/strategy/triplea/delegate/remote/IAbstractPlaceDelegate.java [hunk 1]
```diff
@@ -19,7 +19,6 @@
   @RemoteActionCode(13)
   String placeUnits(Collection<Unit> units, Territory at, BidMode bidMode);
 
-  @RemoteActionCode(12)
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/triplea/delegate/remote/IAbstractPlaceDelegate.java:22]: Technically not needed, this method is only ever invoked in test code, so the annotation is redundant

Comment (ron-murhammer) [reply game-core/src/main/java/games/strategy/triplea/delegate/remote/IAbstractPlaceDelegate.java:22]: Would it be a good idea to add a comment or test annotation to this?

Comment (RoiEXLab) [reply game-core/src/main/java/games/strategy/triplea/delegate/remote/IAbstractPlaceDelegate.java:22]: Comment would be a good idea.

**Comment by DanVanAtta** [comment]
@RoiEXLab thanks for the update to fix the undo problem. I'm wondering if can we be pretty certain we do not have an opcode mismatch problem anywhere else? Is there anyway to find those problems, have you double checked, and do we need to simply rely on manual testing to catch anything that maybe is still an issue?

**Comment by RoiEXLab** [comment]
@DanVanAtta It's pretty simple to identify if there is a problem for a given interface, the list on my gist gives a complete overview of what the "op-codes" should be. Whereever 2 interfaces declare the same method (because of inheritance, which isn't part of the data, but can be checked manually which I did) and have different "op-codes" for it the problem exists.
The only real challenge is the sheer amount of methods that exist, which is the reason I have to automate some of that "identifying" for me, potentially introducing a point of failure.

If you [compare my list](https://github.com/triplea-game/triplea/issues/6104#issuecomment-612032593) with the methods that are actually changed in this PR, you'll notice that there are a lot of methods missing.
The reason for this is that 90% of the methods listed there are actually inherited from (sub)interfaces of `IDelegate`, which don't seem to be used at all over the network. At this point we would've noticed that we needed "op-codes" for them because there are a lot of methods for delegates that would instantly throw an exception, but we haven't seen any exceptions so far so that seems safe.
Another 5% of the methods just happen to have the same signature, but aren't actually related to each other, so no action required there, they already have different codes declared for them.
So what you end up seeing in this PR is the remaining 5% of the methods that could be problematic, after semi-manual review.

TL;DR: There is a risk we could've missed a method. but we can easily re-check if we are unsure.

**Comment by DanVanAtta** [comment]
Anything that can give us full confidence everything works. Manual testing
gives a false sense of security, manual testing is just never going to be
comprehensive enough. If we have 1% of it wrong, we'll have a problem that
is going to be worthy of a rollback, it needs to be right.

**Comment by RoiEXLab** [comment]
@DanVanAtta Thinking about it for a while and there might be a super easy way to check if everything is right.
If we transformed my gist with all the relevant opcodes a little bit so ever line has the format `class:method_name:arg1,arg2,...,argn:code` we could simply verify that all opcodes match (or are simply not present if removed on purpose) via reflection.
What do you think? Would it be worth baking that into a unit test?

**Comment by DanVanAtta** [comment]
I'm not sure how that could be verified by reflection, sounds plausible and that you understand the verification better than I do :grin: 

A unit test to verify that network operations work would be good to have. I built a prototype branch a few nights ago that I'll try to get linked to this PR as soon as I switch computers here...
